### ISSUE TITLE
load.Instances returns _ for JSON entrypoint

### ISSUE THREAD

**Issue by verdverm** [cue-lang/cue#1381]
### What version of CUE are you using (`cue version`)?

v0.4.0

### What did you do?

```
exec go mod tidy
exec go run main.go
cmp stdout golden.stdout

-- main.go --
package main

import (
	"fmt"
	"os"

	"cuelang.org/go/cue/cuecontext"
	"cuelang.org/go/cue/load"
)

func main() {
	ctx := cuecontext.New()

	bis := load.Instances([]string{"a.json"}, nil)

	bi := bis[0]
	if bi.Err != nil {
		fmt.Println("error: ", bi.Err)
		os.Exit(1)
	}

	// Use cue.Context to turn build.Instance to cue.Instance
	value := ctx.BuildInstance(bi)
	if value.Err() != nil {
		fmt.Println("error: ", value.Err())
		os.Exit(1)
	}

	// print json as cue value
	fmt.Println(value)

}
-- a.json --
{
	"a": {
		"b": "B"
	}
}

-- golden.stdout --
a: {
    b: "B"
}
```

### What did you see?

```
$ go run main.go
_
```

### What did you expect to see?

Output similar to `cue export a.json --out cue` which is the `golden.stdout`.

Generally, I would expect a data file explicitly set as an entrypoint to `load.Instances` to be properly parsed and included in a build. This would support JSON, Yaml, and any future codecs like XML, Toml, or ini.

This is the contents of the build.Instance. Notably, the OrphanedFiles seems to be where the json file ends up?

```
&build.Instance{ctxt:(*build.Context)(0xc00036e960), BuildFiles:[]*build.File(nil), IgnoredFiles:[]*build.File(nil), OrphanedFiles:[]*build.File{(*build.File)(0xc0000efd50)}, InvalidFiles:[]*build.File(nil), UnknownFiles:[]*build.File(nil), User:true, Files:[]*ast.File(nil), loadFunc:(build.LoadFunc)(0x8f6860), done:true, PkgName:"", hasName:false, ImportPath:"", Imports:[]*build.Instance(nil), Err:errors.Error(nil), parent:(*build.Instance)(nil), DisplayPath:"command-line-arguments", Module:"", Root:"/home/tony/cue/cuetils", Dir:"/home/tony/cue/cuetils", ImportComment:"", AllTags:[]string(nil), Incomplete:false, ImportPaths:[]string{}, ImportPos:map[string][]token.Pos{}, Deps:[]string{}, DepsErrors:[]error(nil), Match:[]string(nil)}
```

**Comment by verdverm** [comment]
Adding the following is a workaround only for JSON, but not Yaml. Seems the `AddFile` expects CUE, so JSON passes this bar. Should probably be using [AddSyntax](https://pkg.go.dev/cuelang.org/go@v0.4.0/cue/build#Instance.AddSyntax) instead?

```
	...
	
	c, err := os.ReadFile("a.json")
	if err != nil {
		fmt.Println("error: ", err)
		os.Exit(1)
	}

	bi.AddFile("a.json", c)

	// Use cue.Context to turn build.Instance to cue.Instance
	value := ctx.BuildInstance(bi)
	...
```

Ideally, there is a way to set an option on one of the config objects to Context or Load to solve this generally, without having to inspect filenames or globs. This would make it easier to replicate `cue` CLI behavior through the Go API.

I'd like to have entrypoints like `a/*.json` or `b/*.yaml` do what is naively expected.

---

notes: 

Perhaps I can write a function that gets a full filename passed, after glob handling...

- https://pkg.go.dev/cuelang.org/go@v0.4.0/cue/load#Config (set `Context *build.Context` here?)
- https://pkg.go.dev/cuelang.org/go@v0.4.0/cue/build#ParseFile
- https://pkg.go.dev/cuelang.org/go@v0.4.0/cue/build#File
- https://pkg.go.dev/cuelang.org/go@v0.4.0/encoding/yaml#Extract

**Comment by verdverm** [comment]
This seems to be a reasonable workaround


```go
package main

import (
	"fmt"
	"os"

	"cuelang.org/go/cue/ast"
	"cuelang.org/go/cue/cuecontext"
	"cuelang.org/go/cue/load"
	"cuelang.org/go/encoding/json"
	"cuelang.org/go/encoding/yaml"
)

func main() {
	ctx := cuecontext.New()

	bis := load.Instances([]string{"a.json", "b.yaml"}, nil)

	bi := bis[0]
	// check for errors on the instance
	// these are typically parsing errors
	if bi.Err != nil {
		fmt.Println("error: ", bi.Err)
		os.Exit(1)
	}

	for _, f := range bi.OrphanedFiles {
		d, err := os.ReadFile(f.Filename)
		if err != nil {
			fmt.Println("error: ", err)
			os.Exit(1)
		}

		switch f.Encoding {

		case "json":
			A, err := json.Extract(f.Filename, d)
			if err != nil {
				fmt.Println("error: ", err)
				os.Exit(1)
			}

			F := &ast.File{
				Filename: f.Filename,
				Decls:    []ast.Decl{A},
			}
			bi.AddSyntax(F)

		case "yaml":
			F, err := yaml.Extract(f.Filename, d)
			if err != nil {
				fmt.Println("error: ", err)
				os.Exit(1)
			}
			bi.AddSyntax(F)

		default:
			fmt.Println("unknown encoding for", f.Filename, f.Encoding)
		}
	}

	// Use cue.Context to turn build.Instance to cue.Instance
	value := ctx.BuildInstance(bi)
	if value.Err() != nil {
		fmt.Println("error: ", value.Err())
		os.Exit(1)
	}

	// print json as cue value
	fmt.Println(value)

}
```

**Comment by myitcv** [comment]
@verdverm as you will see I started to repurpose this issue as a feature request (because you're running into something that @mpvl and I have discussed before, and it's not the first time that people have expected `cue/load.Instances()` to work like `cmd/cue` does). But I would have had to rewrite your entire description. Instead, I left this issue intact and raised https://github.com/cue-lang/cue/issues/1410. 

For now, this is working as intended.
